manifests/server/dbindex.pp

```diff
@@ -1,24 +1,17 @@
 # See README.md for details.
-define openldap::server::dbindex(
+define openldap::server::dbindex (
   $ensure    = undef,
   $suffix    = undef,
   $attribute = $name,
   $indices   = undef,
 ) {
-
   if ! defined(Class['openldap::server']) {
-    fail 'class ::openldap::server has not been evaluated'
+    fail 'class openldap::server has not been evaluated'
   }
 
-  if $::openldap::server::provider == 'augeas' {
-    Class['openldap::server::install']
-    -> Openldap::Server::Dbindex[$title]
-    ~> Class['openldap::server::service']
-  } else {
-    Class['openldap::server::service']
-    -> Openldap::Server::Dbindex[$title]
-    -> Class['openldap::server']
-  }
+  Class['openldap::server::service']
+  -> Openldap::Server::Dbindex[$title]
+  -> Class['openldap::server']
```

Comment:
Should this be `require => Class['openldap::server::service']` and `before => Class['openldap::server']` on the resource itself?

Reply:
Much more readable, indeed!